### ISSUE TITLE
More colors more types

### ISSUE THREAD

**Issue by tolysz** [bblanchon/SublimeText-HighlightBuildErrors#9]
Hi,
Can we have more colors with more types?
Ideally user configurable ones!

    somefile:302:1: warning: Unknown command "bla"
    somefile:302:1: depricated: Command "bla"
    somefile:302:1: error: Command "bla"
    somefile:302:1: note: Command "bla"
    somefile:302:1: something_to_be_captured_and_converted_into_some_color: Command "bla"

**Comment by bblanchon** [comment]
That would be great!

Here is an idea for the configuration:

```json
{
  "default_color": "invalid",
  "colors": [
    {
      "regex": ".* warning:.*",
      "color": "sublimelinter.mark.error"
    },
    {
      "regex": ".* error:.*",
      "color": "sublimelinter.mark.error"
    }
  ]
}
```

I saw you made a fork. Do you want to implement the feature?

**Comment by tolysz** [comment]
Let me have a stab; it should be not very hard

**Comment by tolysz** [comment]
This is the closes I could do; "warning"... keys are used to distinguish regions 

    {
        "default_color":"invalid",
        "colors":{
        "warning":{
          "regex": ".* warning:.*",
          "color": "sublimelinter.mark.warning"
        },
        "error":{
          "regex": ".* error:.*",
          "color": "sublimelinter.mark.error"
        }
      }
    }

**Comment by bblanchon** [comment]
Good idea.
But why don't you generate the name of the region?
Like `build_errors_color0`, `build_errors_color1`...
I think this would be simpler for the user.

**Comment by tolysz** [comment]
done

**Comment by bblanchon** [comment]
Awesome!
I'll have a look at it while I'm at home and then merge it.

**Comment by bblanchon** [comment]
I made a few changes.

First, there was an exception when the column was not provided by the file_regex.

Then, I changed the settings for something more intuitive:

```js
{
  // the plugin tests each regex and stops at the first match
  "colors": [
    {
      "regex": "(warning|note)",
      "scope": "sublimelinter.mark.warning"
    },
    {
      // default color, when none of the above matches
      "scope": "sublimelinter.mark.error"
    }
  ]
}
```

Note that i simplified the regex and I renamed "color" into "scope" to make it clearer.

The code is pushed in [branch tolysz-issue/9](https://github.com/bblanchon/SublimeText-HighlightBuildErrors/tree/tolysz-issue/9), please give it a try.

**Comment by bblanchon** [comment]
Feature available in version 1.3.0

**Comment by tolysz** [comment]
Thank you :)
